renku/core/management/clone.py

```diff
@@ -33,23 +34,46 @@ def clone(
     skip_smudge=True,
     recursive=True,
     depth=None,
-    progress=None
+    progress=None,
+    config=None,
+    raise_git_except=False,
 ):
     """Clone Renku project repo, install Git hooks and LFS."""
     from renku.core.management.client import LocalClient
 
     path = path or '.'
+
+    if isinstance(path, Path):
+        path = str(path)
+
     # Clone the project
     if skip_smudge:
         os.environ['GIT_LFS_SKIP_SMUDGE'] = '1'
+
     try:
         repo = Repo.clone_from(
             url, path, recursive=recursive, depth=depth, progress=progress
         )
     except GitCommandError as e:
-        raise errors.GitError(
-            'Cannot clone remote Renku project: {}'.format(url)
-        ) from e
+        if not raise_git_except:
+            raise errors.GitError(
+                'Cannot clone remote Renku project: {}'.format(url)
+            ) from e
+
+        raise e
+
+    if config:
+        config_writer = repo.config_writer()
+
+        for key, value in config.items():
+            key_path = key.split('.')
+            if len(key_path) != 2:
```

Comment:
What's the reasoning behind this? There's quite a few git config settings that would have `len(key_path) > 2`, like `filter.lfs.clean=` or `mergetool.meld.cmd`

Reply:
Right, but those are git extensions configuration options which we don't want to have configurable here (cause often we are influencing them via ENV variables - convention I'm happy to break if we decide to change it everywhere). Passed key is in format `<section>.<key>` so in this simple check I'm just making sure that you are setting git only configuration.

Reply:
The reason the LFS flags need to be set via environment variables is because they are used on checkout - afaik it's not possible to set them in the repo config before the repo is cloned/instantiated. For other options that might not be the case.